snuba/query/processors/error_rate_processor.py

```diff
@@ -0,0 +1,59 @@
+from sentry_relay.consts import SPAN_STATUS_NAME_TO_CODE
+
+from snuba.query.conditions import (
+    binary_condition,
+    ConditionFunctions,
+)
+from snuba.query.dsl import count, countIf, div
+from snuba.query.expressions import (
+    Column,
+    Expression,
+    FunctionCall,
+    Literal,
+)
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class ErrorRateProcessor(QueryProcessor):
+    """
+    A percentage of transactions with a bad status. "Bad" status is defined as anything other than "ok" and "unknown".
+    See here (https://github.com/getsentry/relay/blob/master/py/sentry_relay/consts.py) for the full list of errors.
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_functions(exp: Expression) -> Expression:
+            if isinstance(exp, FunctionCall) and exp.function_name == "error_rate":
+                assert len(exp.parameters) == 0
+
+                return div(
+                    countIf(
+                        FunctionCall(
+                            None,
+                            "and",
```

Comment:
you can use `binary_condition` also to build an AND.  And the function to use is declared in BooleanFunctions in the same place as ConditionFunctions